src/Compilers/CSharp/Test/Semantic/Semantics/InterpolationTests.cs

```diff
@@ -5227,5 +5227,950 @@ .locals init (CustomHandler1 V_0)
+                //     public static void M([InterpolatedStringHandlerArgumentAttribute("i")] CustomHandler c, int i) {}
+                Diagnostic(ErrorCode.WRN_ParameterOccursAfterInterpolatedStringHandlerParameter, @"InterpolatedStringHandlerArgumentAttribute(""i"")").WithArguments("int", "CustomHandler").WithLocation(5, 27)
+            );
+
+            CompileAndVerify(comp, sourceSymbolValidator: validate, symbolValidator: validate);
+
+            static void validate(ModuleSymbol module)
+            {
+                var cParam = module.GlobalNamespace.GetTypeMember("C").GetMethod("M").Parameters.First();
+                AssertEx.Equal("System.Runtime.CompilerServices.InterpolatedStringHandlerArgumentAttribute",
+                               cParam.GetAttributes().Single().AttributeClass.ToTestDisplayString());
+                Assert.Equal(1, cParam.InterpolatedStringHandlerArgumentIndexes.Single());
+            }
+        }
+
+        [Fact]
+        public void InterpolatedStringHandlerArgumentAttributeError_MissingConstructor()
+        {
+            var code = @"
+using System.Runtime.CompilerServices;
+class C
+{
+    static void M(int i, [InterpolatedStringHandlerArgumentAttribute(""i"")] CustomHandler c) {}
+}
+";
+
+            var handler = GetCustomHandlerType("CustomHandler", "partial struct", useBoolReturns: true);
+
+            var comp = CreateCompilation(new[] { code, InterpolatedStringHandlerArgumentAttribute, handler }, parseOptions: TestOptions.RegularPreview);
+            comp.VerifyDiagnostics(
+                // (5,27): error CS9011: Could not find a constructor on 'CustomHandler' that matches the parameters specified in the InterpolatedStringHandlerArgumentAttribute.
+                //     static void M(int i, [InterpolatedStringHandlerArgumentAttribute("i")] CustomHandler c) {}
+                Diagnostic(ErrorCode.ERR_CouldNotFindApplicableInterpolatedStringHandlerConstructor, @"InterpolatedStringHandlerArgumentAttribute(""i"")").WithArguments("CustomHandler").WithLocation(5, 27)
+            );
+
+            var cParam = comp.SourceModule.GlobalNamespace.GetTypeMember("C").GetMethod("M").Parameters.Skip(1).Single();
+            AssertEx.Equal("System.Runtime.CompilerServices.InterpolatedStringHandlerArgumentAttribute",
+                           cParam.GetAttributes().Single().AttributeClass.ToTestDisplayString());
+            Assert.Empty(cParam.InterpolatedStringHandlerArgumentIndexes);
+        }
+
+        [Fact]
+        public void InterpolatedStringHandlerArgumentAttributeError_InaccessibleConstructor()
+        {
+            var code = @"
+using System.Runtime.CompilerServices;
+class C
+{
+    static void M(int i, [InterpolatedStringHandlerArgumentAttribute(""i"")] CustomHandler c) {}
+}
+
+public partial struct CustomHandler
+{
+    private CustomHandler(int literalLength, int formattedCount, int i) : this() {}
```

Comment:
>private CustomHandler(int literalLength, int formattedCount, int i) : this() {} [](http://example.com/codeflow?start=4&length=79)

Accessibility is another aspect that is location specific. For example, internal/protected constrcutor might be accessible from where M will be called.